src/CabalHelper/Runtime/Main.hs

```diff
@@ -547,6 +640,10 @@ componentEntrypoints (CLib Library {..})
     = ChLibEntrypoint
         (map gmModuleName exposedModules)
         (map gmModuleName $ otherModules libBuildInfo)
+#if CH_MIN_VERSION_Cabal(2,0,0)
+componentEntrypoints (CFLib (ForeignLib{..}))
+    = ChLibEntrypoint [] []
```

Comment:
You do have to fill out the `chOtherModules` at least. It's easy just use `otherModules` and `foreignLibBuildInfo`.